app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala

```diff
@@ -60,7 +60,7 @@ package object types extends SimpleJson {
   type TraderRef = String @@ TraderRef.Tag
   object TraderRef
       extends RegexValidatedString(
-        regex = """.{0,35}"""
+        regex = """.{1,400}"""
```

Comment:
This now matches the frontend validation else an empty string will blow the front end up.

Really there needs to be a single source of truth.

Reply:
Patrick,  in my latest PR  the regex is actually updated to the folowing...

 regex = """[A-Za-z0-9 ]{1,400}"""

https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/357/files#diff-dd0d4530e38e33f61c2e5d4d3a367c1cbbe356c51c478e4e2d68eefaa73d0344

Reply:
@masahmad, I think `\s` is what is used for representing whitespace. Also Mas' regex allows whitespace/numbers/letters but no special characters. @pbyrne84, your regex allows any characters. We should confirm (from ETMP) if special characters are allowed and out frontend and our backend validation should match.